app/model/application.js

```diff
@@ -1,241 +1,445 @@
-            }
-          }
-          if (buildsFinished) {
-            if (buildsFailed) {
-              // TODO: Builds Failed, File issues on GitHub
-              iteration.status = 'FAILED';
-            } else {
-              iteration.status = 'SUCESS'
-            }
-          }
-          application.save()
-        });
-    });
+// Updates all Github data from database
+// Returns promise of same application on success
+ApplicationSchema.methods.githubPush = function() {
+  const application = this;
+  const repoName = application.github.name;
+  const fullName = application.github.fullName;
+  app.log.silly(`application githubPush called for ${fullName}`);
+
+  return application.labelPush();
+}
+
+// Updates all database data from Github and Github from database data
+// Returns saved application including 'errors' always
+ApplicationSchema.methods.githubSync = function() {
+  let application = this;
+  const repoName = application.github.name;
+  const fullName = application.github.fullName;
+  app.log.silly(`application githubSync called for ${fullName}`);
+
+  return application.githubFetch()
+  .then(Application.githubPush); // No need to resave, as push doesn't save anything
 }
 
-ApplicationSchema.statics.debianChangelog = function(application, params) {
-  const promises = [];
-  for (let i in params) {
+// Other methods
+// TODO: add release target for changelog generation (default to latest)
+// Generates changelog
+// Returns array of promises of generated changelogs
+ApplicationSchema.methods.changelog = function(params) {
+  const application = this;
+  const fullName = application.github.fullName;
+  app.log.silly(`application changelog called for ${fullName}`);
+
+  let promises = [];
+  for (let i = 0; i < params.length; i++) {
     promises.push(new Promise((resolve, reject) => {
-      app.render('debian-chlg', {
-        layout:       false,
-        dist:         params[i].DIST,
-        package:      params[i].PACKAGE,
-        releases:   application.releases,
-      }, (err, changelog) => {
-        if (err) {
-          reject(err);
-        } else {
-          params[i].CHANGELOG = changelog;
-          resolve(params[i]);
+      app.render('changelog', {
+        layout:   false,
+        dist:     params[i].DIST,
+        package:  params[i].PACKAGE,
+        releases: application.releases,
+      }, (error, changelog) => {
+        if (error) {
+          return reject(error);
         }
-      })
+
+        params[i].CHANGELOG = changelog;
+        return resolve(params[i]);
+      });
     }));
   }
+
   return promises;
 }
 
-ApplicationSchema.statics.syncIssuesToGitHub = function(application) {
-  return Promise
-    .all(application.issues.map(issue => issue.syncToGitHub()))
-    .then(() => application);
-}
+// Other statics
+// Updates build given update data
+// Returns promise of saved build
+ApplicationSchema.statics.buildUpdate = function(data) {
+  const split = data.IDENTIFIER.split('#');
```

Comment:
can you include an example above this line of what `data.IDENTIFIER` looks like? and/or, could you use destructuring assignment to assign the components of the split into specific named variables?